### ISSUE TITLE
Squeezenet on Mali and OpenCL crashes

### ISSUE THREAD

**Issue by ehsanmok** [dmlc/nnvm#490]
Hi

Compiling the squeezenet (pretrained from gluon) compiles for OpenCL but it crashes on android with `CL_OUT_OF_RESOURCES` despite changing the nnvm config build `with tvm.build_config` unrolling parameters etc. 

However, it doesn't even compile for Mali with errors (with input 256x256 and the same with 224x224):

```
graph = graph.apply("GraphFusePartition").apply("GraphFuseCompile")
  File "/home/ilab/anaconda3/lib/python3.6/site-packages/nnvm-0.8.0-py3.6.egg/nnvm/graph.py", line 234, in apply
    check_call(_LIB.NNGraphApplyPasses(self.handle, npass, cpass, ctypes.byref(ghandle)))
  File "/home/ilab/anaconda3/lib/python3.6/site-packages/nnvm-0.8.0-py3.6.egg/nnvm/_base.py", line 75, in check_call
    raise NNVMError(py_str(_LIB.NNGetLastError()))
nnvm._base.NNVMError: TVMCall CFunc Error:
Traceback (most recent call last):
  File "tvm/_ffi/_cython/./function.pxi", line 38, in core.tvm_callback
  File "/home/ilab/anaconda3/lib/python3.6/site-packages/nnvm-0.8.0-py3.6.egg/nnvm/top/nn.py", line 99, in compute_conv2d
    out = topi.nn.conv2d(inputs[0], kernel, strides, padding, layout)
  File "<decorator-gen-33>", line 2, in conv2d
  File "/home/ilab/anaconda3/lib/python3.6/site-packages/tvm-0.2.0-py3.6-linux-x86_64.egg/tvm/target.py", line 343, in dispatch_func
    return dispatch_dict[k](*args, **kwargs)
  File "/home/ilab/anaconda3/lib/python3.6/site-packages/topi-0.2.0-py3.6.egg/topi/mali/conv2d.py", line 128, in decl_conv2d
    return _decl_spatialpack(data, kernel, stride, padding, layout, out_dtype)
  File "/home/ilab/anaconda3/lib/python3.6/site-packages/topi-0.2.0-py3.6.egg/topi/mali/conv2d.py", line 215, in _decl_spatialpack
    assert OW % VW == 0, "OW: %d  VW : %d" % (OW, VW)
AssertionError: OW: 127  VW : 4

```
I couldn't decode what those OW, VW parameters and figure out the assertion logic. 
Any idea?

**Comment by merrymercy** [comment]
Current mali convolution schedules only guarantee good performance for common convolution shapes in resnet, vgg, mobilenet. So I added this assertion for untested shape configuration.

`OW`, `VW` are the tiling factors on some dimensions, they should be tuned according to the shape configurations to get the best performance.

If you want to check the correctness, you can insert these lines before this assertion ("topi/mali/conv2d.py", line 215)
```python
if OW % VW != 0:
    while OW % VW != 0:
        VW //= 2
    if CO % (2 * VC) == 0:
        VC *= 2
```

If you want to get good performance, you should tune these parameters. Or you can try another schedule algorithm im2col, by changing the if conditions here https://github.com/dmlc/tvm/blob/64b3d19fffdb99e211ff3408b504ed06bc9b7f00/topi/python/topi/mali/conv2d.py#L122

**Comment by merrymercy** [comment]
What's your target gpu?

OpenCL will use the cuda's schedule, which is tuned on NVIDIA server class gpu. It may use too many threads in a block or too much shared/local memory.

related #530

**Comment by ehsanmok** [comment]
@merrymercy target is for Mali T830 and in build `-target=arm64-linux-android`. Actually I just removed the Mali assertion above then could compile Squeezenet (with opt_level=0) for Mali and it works on the phone!

For OpenCL the target gpu is Mali G71, I already tried tweaking parameters in `tvm.build_config` with no success. I think it's because of high img size 400x400ish causes insufficient memory!
